wandb/util.py

```diff
@@ -896,7 +898,33 @@ def check_retry_commit_artifact(e: Any) -> bool:
         and e.response.status_code == 409
     ):
         return True
-    return no_retry_auth(e)
+    return None
+
+
+def make_check_retry_fn(
+    fallback_retry_fn: CheckRetryFnType,
+    check_fn: Callable[[Exception], Optional[bool]],
```

Comment:
while i agree, the tristate is a bit confusing.  I have added some docstrings.

I think keeping it as a tristate makes sense because it is symmetric with `check_retry_fn` and allows composability of check functions with the optional.

```
    Returns:
        True - Should retry this operation
        False - Should not retry this operation
        None - No decision, let someone else decide
```